velox/connectors/hive/HiveConnectorSplit.h

```diff
@@ -43,7 +44,8 @@ struct HiveConnectorSplit : public connector::ConnectorSplit {
           _partitionKeys = {},
       std::optional<int32_t> _tableBucketNumber = std::nullopt,
       const std::unordered_map<std::string, std::string>& _customSplitInfo = {},
-      const std::shared_ptr<std::string>& _extraFileInfo = {})
+      const std::shared_ptr<std::string>& _extraFileInfo = {},
+      const std::shared_ptr<dwio::common::ReaderOptions>& readerOptions = {})
```

Comment:
`const std::shared_ptr<dwio::common::ReaderOptions> readerOptions = std::make_shared<dwio::common::ReaderOptions>()` is probably cleaner since this cannot be nullptr. We can then `std::move` to optimize.

`if (!options)` checks can be avoided.

Reply:
Done.